src/JustEat.StatsD/StatsDTcpClient.cs

```diff
@@ -0,0 +1,127 @@
+            {
+                _ipBasedEndpoint = new IPEndPoint(address, _port);
+            }
+        }
+
+        public bool Send(string metric)
+        {
+            return Send(new[] { metric });
+        }
+
+        [SuppressMessage("Microsoft.Design", "CA1031:DoNotCatchGeneralExceptionTypes", Justification = "This is one of the rare cases where eating exceptions is OK")]
+        public bool Send(IEnumerable<string> metrics)
+        {
+            var data = EventArgsPool.Pop();
+            //firehose alert! -- keep it moving!
+            if (null == data)
+            {
+                return false;
+            }
+
+            try
+            {
+                data.RemoteEndPoint = GetIPEndPoint();
+                data.SendPacketsElements = metrics.ToMaximumBytePackets()
+                    .Select(bytes => new SendPacketsElement(bytes, 0, bytes.Length, true))
+                    .ToArray();
+
+                using (var tcpClient = GetTcpClient())
+                {
+                    tcpClient.Client.Connect(data.RemoteEndPoint);
+                    tcpClient.Client.SendPacketsAsync(data);
+                }
+
+                Trace.TraceInformation("statsd: {0}", string.Join(",", metrics));
+
+                return true;
+            }
+            //fire and forget, so just eat intermittent failures / exceptions
+            catch (Exception e)
+            {
+                Trace.TraceError("General Exception when sending metric data to statsD :- Message : {0}, Inner Exception {1}, StackTrace {2}.", e.Message, e.InnerException, e.StackTrace);
+            }
+
+            return false;
+        }
+
+        public TcpClient GetTcpClient()
+        {
+            TcpClient client = null;
+            try
+            {
+                client = new TcpClient
+                {
+                    Client = { SendBufferSize = 0 }
+                };
+            }
+            catch (SocketException e)
+            {
+                Trace.TraceError(string.Format(CultureInfo.InvariantCulture, "Error Creating tcpClient :-  Message : {0}, Inner Exception {1}, StackTrace {2}.", e.Message, e.InnerException, e.StackTrace));
+            }
+            return client;
+        }
+
+        /// <summary>	Performs application-defined tasks associated with freeing, releasing, or resetting unmanaged resources. </summary>
+        public void Dispose()
+        {
+            if (!_disposed)
+            {
+                Dispose(true);
+                GC.SuppressFinalize(this);
+            }
+        }
+
+        /// <summary>	Performs application-defined tasks associated with freeing, releasing, or resetting unmanaged resources. </summary>
+        /// <param name="disposing">	true if resources should be disposed, false if not. </param>
+        protected virtual void Dispose(bool disposing)
+        {
+            if (disposing)
+            {
+                _disposed = true;
+            }
+        }
```

Comment:
Do we need all this for disposal? the method is essentially empty. can you Delete it or reduce it to `{ }`. 
Also the comments "Performs application-defined tasks..." are vague boilerplate, remove them